tensorflow/python/training/experimental/mixed_precision.py

```diff
@@ -73,39 +73,96 @@ def _wrap_optimizer(opt, loss_scale):
+  are considered or rejected by this rewrite operation, and they are defined in
+  [auto_mixed_precision_lists.h](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h),
+  and each category is summarized below:
+
+  * [`ClearList`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h#L178): Ops that do not have numerically significant effects. E.g. `ArgMax` and `Floor`.
+  * [`WhiteList`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h#L55): Ops that are considered numerically safe for execution in float16, and are always converted. E.g. `Conv2D`.
+  * [`GrayList`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h#L92): Ops that are considered numerically safe for execution in float16 unless downstream from a BlackList Op. E.g. `Add` and `AvgPool`.
+  * [`BlackList`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h#L146): Ops that are numerically dangerous to execute in float16 and can negatively affect downstream nodes. E.g. `Softmax`.
 
   When this function is used, gradients should only be computed and applied with
-  the returned optimizer, either by calling `opt.minimize()` or
-  `opt.compute_gradients()` followed by `opt.apply_gradients()`. Gradients
-  should not be computed with `tf.gradients` or `tf.GradientTape`. This is
-  because the returned optimizer will apply loss scaling, and
-  `tf.gradients`/`tf.GradientTape` will not. If you do directly use
-  `tf.gradients` or `tf.GradientTape`, your model may train to a worse quality.
-
+  the returned optimizer, either by calling `opt.minimize()` or `opt.compute_gradients()`
+  followed by `opt.apply_gradients()`. Gradients should not be computed with
+  `tf.gradients` or `tf.GradientTape`. This is because the returned optimizer will
+  apply loss scaling, and `tf.gradients` or `tf.GradientTape` will not. If you do
+  directly use `tf.gradients` or `tf.GradientTape`, your model may not converge
+  due to float16 underflow or overflow problems.
+  
   When eager execution is enabled, the mixed precision graph rewrite is only
   enabled within `tf.function`s, as outside `tf.function`s, there is no graph.
 
-  When enabled, mixed precision is only used on Volta GPUs and above. The parts
+  **Performance Guide**
+
+  * Dimensions should be in multiples of 8 where possible.
+  * Dimensions should at least 64, and where smaller than 256, be a power of 2.
+  * Operations with dimensions less than 128 are memory-bound rather than compute-bound.
+  * Batch size should be a multiple of SM count (e.g. a multiple of 80 on V100 GPUs).
+  * Where needed, padding the tensor to a multiple of 8 will result in better performance.
```

Comment:
put this line second (or third), so it comes together with the other bullets about tensor dimensions. right now its after batch size so is a little confusing.